app/controllers/challenges_controller.rb

```diff
@@ -0,0 +1,57 @@
+class ChallengesController < ApplicationController
+  before_action :set_and_authorize_challenge, only: [ :show, :edit, :update, :destroy ]
+
+  def index
+    @challenges = Challenge.all
+    authorize(@challenges)
+  end
+
+  def show
+  end
+
+  def new
+    @challenge = Challenge.new
+    authorize(@challenge)
+  end
+
+  def create
+    @challenge = Challenge.new(challenge_params.merge(owner: Current.user))
```

Comment:
```suggestion
    @challenge = Current.user.challenges.build(challenge_params)
```

This will require adding the missing:

```ruby
class User
  has_many :challenges, inverse_of: :owner, dependent: destroy
  #...
end
```